File: hospital/SergioDeIscar-hospital-java/SergioDeIscar-hospital-java/src/main/java/org/example/interfaces/HospitalExtension.java
package org.example.interfaces;

import org.example.enums.TipoPaciente;
import org.example.models.Paciente;

import java.util.List;

public interface HospitalExtension extends CrudRepository<Paciente, String> {
    Boolean estaCompleto();
    int numPacientes();

    List<Paciente> pacientesOrdeFechaIngreso();
    List<Paciente> pacientesOrdeNombre();
    List<Paciente> pacientesOrdeDNI();

    List<Paciente> pacientesPorTipo(TipoPaciente tipo);
    int numPacientePorTipo(TipoPaciente tipo);
    int capacidadMaxima();
}

File: hospital/SergioDeIscar-hospital-kotlin/SergioDeIscar-hospital-kotlin/src/main/kotlin/interfaces/HospitalExtension.kt
package interfaces

import enums.TipoPaciente
import models.Paciente

interface HospitalExtension: CrudRepository<Paciente, String> {
    fun estaCompleto(): Boolean
    fun numPacientes(): Int

    fun pacientesOrdeFechaIngreso(): List<Paciente>
    fun pacientesOrdeNombre(): List<Paciente>
    fun pacientesOrdeDNI(): List<Paciente>

    fun pacientesPorTipo(tipo: TipoPaciente): List<Paciente>
    fun numPacientePorTipo(tipo: TipoPaciente): Int
}